src/oci-umount.c

```diff
@@ -342,12 +343,26 @@ static int prestart(const char *rootfs,
 		return EXIT_FAILURE;
 	}
 
+	snprintf(process_pid_ns_fd, PATH_MAX, "/proc/%d/ns/pid", pid);
+
+	pfd = open(process_pid_ns_fd, O_RDONLY);
+	if (pfd < 0) {
+		pr_perror("Failed to open pid namespace fd %s", process_pid_ns_fd);
+		return EXIT_FAILURE;
+	}
+
 	/* Join the mount namespace of the target process */
 	if (setns(fd, 0) == -1) {
 		pr_perror("Failed to setns to %s", process_mnt_ns_fd);
 		return EXIT_FAILURE;
 	}
 
+	/* Join the pid namespace of the target process */
+	if (setns(pfd, 0) == -1) {
```

Comment:
No that is just to get the current process PID into the pid namespace, as soon as this call is done, I can read /proc/1/mountinfo.